sci_strategic_plugin/src/sci_strategic_plugin.cpp

```diff
@@ -394,4 +397,43 @@ double SCIStrategicPlugin::findSpeedLimit(const lanelet::ConstLanelet& llt) cons
   }
 }
 
+cav_msgs::MobilityOperation SCIStrategicPlugin::generateMobilityOperation()
+{
+    cav_msgs::MobilityOperation mo_;
+    mo_.header.timestamp = ros::Time::now().toNSec();
+    mo_.header.sender_bsm_id = std::string(reinterpret_cast<const char*>(getMsgId(ros::Time::now())[0]), getMsgId(ros::Time::now()).size());
+
+    int flag = (approaching_stop_controlled_interction_ ? 1 : 0);
+
+    double vehicle_acceleration_limit_ = config_.vehicle_accel_limit * config_.vehicle_accel_limit_multiplier;
+    double vehicle_deceleration_limit_ = -1 * config_.vehicle_decel_limit * config_.vehicle_decel_limit_multiplier;
+
+    mo_.strategy_params = "intersection_box_flag, acceleration_limit, deceleration_limit," + std::to_string(flag) + "," + std::to_string(vehicle_acceleration_limit_) + "," + std::to_string(vehicle_deceleration_limit_);
+
+    mobility_operation_pub.publish(mo_);
+}
+
+std::vector<uint8_t> SCIStrategicPlugin::getMsgId(const ros::Time now)
```

Comment:
As mentioned above, I dont think we should generate bsm id ourself. We need to subscribe to the bsm msg and extract the id from that. The bsm id is how the status and bsm messages can be matched,